### PR TITLE
[#1063] Updated helm chart to support multi-az

### PR DESCRIPTION
**Pull request by Arnav15**
Added nodeAffinity spec to orchestrate pod deployments on the nodes in the required zones. Also updated the master addresses to be customizable for multi-az where the addresses can't be computed by the template.
Adds support for issue #1063

### CODE REVIEW

File: cloud/kubernetes/helm/yugabyte/templates/service.yaml [hunk 1]
```diff
@@ -212,8 +224,13 @@ spec:
           - "--fs_data_dirs={{ template "yugabyte.fs_data_dirs" $storageInfo }}"
           - "--rpc_bind_addresses={{ template "yugabyte.server_address" . }}"
           - "--server_broadcast_addresses={{ template "yugabyte.server_address" . }}:7100"
+          {{ if $root.Values.isMultiAz }}
+          - "--master_addresses={{ $root.Values.masterAddresses}}"
+          - "--replication_factor={{ $root.Values.replicas.totalMasters }}"
```
Comment (rkarthik007) [inline cloud/kubernetes/helm/yugabyte/templates/service.yaml]: You would need the masters to come up with the correct placement info. See this step: https://docs.yugabyte.com/latest/deploy/public-clouds/aws/#create-yb-master1-s-config-file

There should be a corresponding thing done for the tservers also.

File: cloud/kubernetes/helm/yugabyte/values.yaml [hunk 2]
```diff
@@ -60,6 +60,8 @@ PodManagementPolicy: Parallel
 
 enableLoadBalancer: True
 
+isMultiAz: False
```
Comment (rkarthik007) [inline cloud/kubernetes/helm/yugabyte/values.yaml]: If this is set, at the end of the deploy you need to set the config in the master also, its an API call. See this step: https://docs.yugabyte.com/latest/deploy/public-clouds/aws/#7-configure-az-region-aware-placement

File: cloud/kubernetes/helm/yugabyte/templates/service.yaml (hunk 1 above)
Comment (Arnav15) [reply cloud/kubernetes/helm/yugabyte/templates/service.yaml]: Those flags are passed while creating the universe through YW.

File: cloud/kubernetes/helm/yugabyte/values.yaml (hunk 2 above)
Comment (Arnav15) [reply cloud/kubernetes/helm/yugabyte/values.yaml]: I believe this is also done through YW during the create universe step.

File: cloud/kubernetes/helm/yugabyte/templates/service.yaml (hunk 1 above)
Comment (rkarthik007) [reply cloud/kubernetes/helm/yugabyte/templates/service.yaml]: Got it!

**Comment by rkarthik007** [comment]
This mostly looks good to me, will let @ramkumarvs take a look and accept.

@Arnav15 - can you also write the set of steps you used to being up a multi-zone universe? For example, creating the storage classes, args for each run of the helm chart, etc?

**Pull request merged**